power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -714,52 +766,174 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
   public:
     template <class Component> using UpdateType = typename Component::UpdateType;
 
-    static bool is_update_independent(ConstDataset const& update_data) {
-        // If the batch size is (0 or) 1, then the update data for this component is 'independent'
-        if (update_data.batch_size() <= 1) {
-            return true;
-        }
+    UpdateCompIndependence check_components_independence(ConstDataset const& update_data) const {
+        auto const all_comp_count_in_base = this->all_component_count();
+        Idx const n_scenarios = update_data.batch_size();
+
+        auto check_ids_na = [](auto const& all_spans) {
```

Comment:
we definitely need to profile this function because it is something that causes a pre-process along all scenarios w/ copy (rather than looping once and accumulating on the fly)

Reply:
Resolved in https://github.com/PowerGridModel/power-grid-model/pull/818